nucypher/crypto/kits.py

```diff
@@ -76,3 +80,39 @@ def __init__(self, *args, **kwargs) -> None:
     def from_bytes(cls, some_bytes):
         capsule, sender_pubkey_sig, ciphertext = cls.split_bytes(some_bytes)
         return cls(capsule=capsule, sender_pubkey_sig=sender_pubkey_sig, ciphertext=ciphertext)
+
+
+class RevocationKit:
+
+    def __init__(self, policy_treasure_map, signer: 'SignatureStamp'):
+        from nucypher.policy.models import Revocation
+        self.revocations = dict()
+        for node_id, arrangement_id in policy_treasure_map:
+            self.revocations[node_id] = Revocation(arrangement_id, signer=signer)
+
+    def __iter__(self):
+        return iter(self.revocations.values())
+
+    def __getitem__(self, node_id):
+        return self.revocations[node_id]
+
+    def __len__(self):
+        return len(self.revocations)
+
+    def __eq__(self, other):
+        return self.revocations == other.revocations
+
+    @property
+    def revokable_addresses(self):
+        """
+        Returns a Set of revokable addresses in the checksum address formatting
+        """
+        return set(self.revocations.keys())
+
+    def add_confirmation(self, node_id, signed_receipt):
+        """
+        Adds a signed confirmation of Ursula's ability to revoke the arrangement.
+        """
+        # TODO: Verify Ursula's signature
+        # TODO: Implement receipts
+        raise NotImplementedError
```

Comment:
So - Ideas here: How about a `dict` of arrangements to ... .  Also - do we want to persist these to disk?